src/zenml/integrations/github/code_repositories/github_code_repository.py

```diff
@@ -84,8 +118,16 @@ def login(
         """
         try:
             self._github_session = Github(self.config.token)
-            user = self._github_session.get_user().login
-            logger.debug(f"Logged in as {user}")
+            if self.config.token:
+                user = self._github_session.get_user().login
+                logger.debug(f"Logged in as {user}")
+            else:
+                logger.info(
+                    "No token provided. Access is possible to public repositories only."
+                )
+                self.check_github_repo_public(
```

Comment:
Tbh I am not sure this is the right workflow. When the traceback hits, the log "No token provided. Access is possible to public repositories only." will be lost. Better to throw an error like "It is not possible to access this repository as it does not appear to be public. Access to private repositories is only possible when a token is provided. Please provide a token and try again"